File: tests/Generator/resource/csharp/complex/StringPropertyType.cs
using System.Text.Json.Serialization;

namespace TypeAPI.Model;

/// <summary>
/// Represents a string value
/// </summary>
public class StringPropertyType : ScalarPropertyType
{
    [JsonPropertyName("default")]
    public string? Default { get; set; }

    [JsonPropertyName("format")]
    public string? Format { get; set; }

    [JsonPropertyName("type")]
    public new string? Type { get; set; } = "string";

}

File: tests/Generator/resource/rust/complex/string_property_type.rs
use serde::{Serialize, Deserialize};
use scalar_property_type::ScalarPropertyType;

// Represents a string value
#[derive(Serialize, Deserialize)]
pub struct StringPropertyType {
    #[serde(rename = "deprecated")]
    deprecated: Option<bool>,

    #[serde(rename = "description")]
    description: Option<String>,

    #[serde(rename = "nullable")]
    nullable: Option<bool>,

    #[serde(rename = "type")]
    _type: Option<String>,

    #[serde(rename = "format")]
    format: Option<String>,

}

